internal/credit/postgres_connector/ledger.go

```diff
@@ -0,0 +1,110 @@
+package postgres_connector
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/oklog/ulid/v2"
+
+	"github.com/openmeterio/openmeter/internal/credit"
+	"github.com/openmeterio/openmeter/internal/credit/postgres_connector/ent/db"
+	db_ledger "github.com/openmeterio/openmeter/internal/credit/postgres_connector/ent/db/ledger"
+	"github.com/openmeterio/openmeter/internal/credit/postgres_connector/ent/pgulid"
+	"github.com/openmeterio/openmeter/pkg/slicesx"
+)
+
+func (c *PostgresConnector) CreateLedger(ctx context.Context, namespace string, ledgerIn credit.Ledger, upsert bool) (credit.Ledger, error) {
+	dbLedger, err := transaction(ctx, c, func(tx *db.Tx) (*db.Ledger, error) {
```

Comment:
have you considered always calling upsert and using OnConflict Do nothing

Reply:
For the moment I would stay with the traditional REST schematics where we are not upserting anything. This API can be easily updated to support upserts in a non-breaking manner, however if we implement something right now we might need to do a breaking change in the future when we better understand the problem space for ledgers.

I don't see how we want to handle account deletion/closure and having an upsert call might get convoluted if we start doing things like account status, etc. I think we will see this as we are starting to address more and more use-cases. 

Also this API changes the expected behavior too (so I don't think we have value (at least for now) in upserting). If there's an external system we are doing the Entitlements for, what will happen is that:
- The external system has its own account database
- When needed it creates the ledger in our system, persists the ledger id into their account database
- They do grants based on this ID

So the Create Ledger will most probably called only on account creation. 

I might be wrong, but I think we should see actual use-cases both for upsert and account closure before adding specific APIs for there. (And we can add that if we want later on).